### PR TITLE
fix: keep unsubscribed donors as isDonating

### PR DESCRIPTION
**Pull request by ahmaxed**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] All the files I changed are in the same world language, for example: only English changes, or only Chinese changes, etc.

@raisedadead, this pr updates the logic for Paypal unsubscription hooks, so only the donation record is updated and the user records stays as `isDonating`;

### CODE REVIEW

File: api-server/server/utils/donation.js [hunk 1]
```diff
@@ -160,10 +160,7 @@ export async function cancelDonation(body, app) {
 
     User.findOne({ where: { id: userId } }, (err, user) => {
       if (err || !user || !user.donationEmails) throw Error(err);
-      log('Updating user record for donation cancellation');
-      user.updateAttributes({
-        isDonating: false
-      });
+      log('Updating donation record for donation cancellation');
     });
```
Comment (ojeytonwilliams) [inline api-server/server/utils/donation.js:164]: Can't we simply remove this update entirely?

File: api-server/server/utils/donation.test.js [hunk 2]
```diff
@@ -157,10 +156,6 @@ describe('donation', () => {
       expect(updateDonationAttr).toHaveBeenCalledWith({
         endDate: new Date(status_update_time).toISOString()
       });
-
-      expect(updateUserAttr).toHaveBeenCalledWith({
-        isDonating: false
-      });
```
Comment (ojeytonwilliams) [inline api-server/server/utils/donation.test.js:163]: Rather than remove this, how about just using `.not.toHaveBeenCalled()`?

**Review by ojeytonwilliams — CHANGES_REQUESTED** [review]

File: api-server/server/utils/donation.js (hunk 1 above)
Comment (raisedadead) [reply api-server/server/utils/donation.js:164]: Yup - I do not see why make an expensive call here if the goal is not to set anything.

Comment (ahmadabdolsaheb) [reply api-server/server/utils/donation.js:164]: That makes sense.

**Review by ojeytonwilliams — APPROVED** [review]
LGTM :+1:

**Pull request merged**